tree/dataframe/src/RDataFrame.cxx

```diff
@@ -1520,6 +1526,69 @@ Examples of implemented progress bars can be seen by running [Higgs to Four Lept
 */
 // clang-format on
 
+namespace {
+
+std::shared_ptr<ROOT::Detail::RDF::RLoopManager> CreateLMFromTTree(std::string_view datasetName,
```

Comment:
if the above refactoring idea pans out these could become a single constructor in RLoopManager, e.g. `RLoopManager(EInputType, datasetName, fileNameGlob, defaultColumns)` with `EInputType` being an `enum class` with values `kTTree`, `kRNTuple`, `kInferred` or similar.

Reply:
For now I will keep the functions separate and maybe converge on a single constructor in a separate PR